src/api/java/com/minecolonies/api/colony/buildings/IBuilding.java

```diff
@@ -32,6 +34,27 @@
 
 public interface IBuilding extends IBuildingContainer, IRequestResolverProvider, IRequester
 {
+    /**
+     * Check if the building has a particular module.
+     * @param clazz the module of the class to check.
+     * @return true if so.
+     */
+    boolean hasModule(final Class<? extends AbstractBuildingModule> clazz);
```

Comment:
No, this should be `Class<? extends IBuildingModule>` or what ever the root interface for a module is.